picoquic/picosocks.c

```diff
@@ -641,8 +645,15 @@ void picoquic_socks_cmsg_format(
             }
 #else 
             /* The IP_PKTINFO structure is not defined on BSD */
+            /* Some versions of freeBSD do not define IP_SENDSRCADDR, use IP_RECVDSTADDR instead. */
             struct in_addr* pktinfo = (struct in_addr*)cmsg_format_header_return_data_ptr(msg, &last_cmsg,
-                &control_length, IPPROTO_IP, IP_SENDSRCADDR, sizeof(struct in_addr));
+                &control_length, IPPROTO_IP,
+#ifdef IP_SENDSRCADDR
+                IP_SENDSRCADDR
+#else
+                IP_RECVDSTADDR
+#endif
```

Comment:
I added a comment so we understand the motivation for the #ifdef, might be useful for maintenance.

Reply:
Thank you! (This is also true for some versions of macOS.)